### ISSUE TITLE
Incorrect seek to function

### ISSUE THREAD

**Issue by Kyant0** [libxmp/libxmp#734]
The seek function is always "lag/delay" about 6 seconds.
For example, when seek to 3s, it will play at 0s; when  seek to end, it will play at (end - 6s).

I tried OpenMPT and modplug, they don't have the problem.

**Comment by AliceLR** [comment]
This is because `xmp_seek_time` only seeks to the start of a pattern, which at the default MOD tempo is about 6 seconds long. For finer grained seeking, we would need to add a second function that performs `xmp_seek_time` followed by processing several frames until it reaches the closest frame possible. (I'm not sure why libxmp wasn't doing this in the first place.)

**Comment by mmontag** [comment]
Yeah. We need a real seek function, even if it's a little expensive. Rename the issue?

In the meantime, this limitation should probably be called out in the documentation.

**Comment by mmontag** [comment]
Another quirk about this method: when you seek to a time that falls within the *currently playing* pattern, playback is not restarted at the beginning of the pattern; the seek is ignored.  This behavior is confusing, as you can see above.

**Comment by mmontag** [comment]
Could this be assigned to a specific version milestone? Don't want to lose sight of this one.
